### PR TITLE
NEWRELIC-6379: (config) enable otlp_endpoint config item

### PR DESCRIPTION
**Pull request by jordigh**
## Links

* NEWRELIC-6379

## Details

The agent seems like the right place to have this sort of logic, similar to the logic for hitting the ordinary NR collector endpoint.

### CODE REVIEW

File: test/unit/config/config-env.test.js [hunk 1]
```diff
@@ -115,6 +115,25 @@ tap.test('when overriding configuration values via environment variables', (t) =
     })
   })
 
+  t.test('should parse the region off the license key for OTel', (t) => {
```
Comment (bizob2828) [inline test/unit/config/config-env.test.js:118]: we're also missing a test where a region cannot be parsed

File: lib/config/index.js [hunk 2]
```diff
@@ -1298,14 +1298,21 @@ Config.prototype._canonicalize = function _canonicalize() {
...
   if (this.host === '') {
-    const region = parseKey(this.license_key)
     if (region) {
-      this.host = 'collector.' + region + '.nr-data.net'
+      this.host = `collector.${region}.nr-data.net`
     } else {
       this.host = 'collector.newrelic.com'
     }
   }
+  if (this.otlp_endpoint === '') {
```
Comment (bizob2828) [inline lib/config/index.js:1309]: cool so to point this to staging we'd have to set `NEW_RELIC_OTLP_ENDPOINT`?

File: lib/config/default.js [hunk 3]
```diff
@@ -52,6 +52,13 @@ defaultConfig.definition = () => ({
    * You shouldn't need to change this.
    */
   host: '',
+  /**
+   * Endpoint to send OpenTelemetry spans to.
+   *
+   * This should be automatically deduced from your region and other
+   * settings, but if desired, you can override it.
+   */
+  otlp_endpoint: '',
```
Comment (bizob2828) [inline lib/config/default.js:61]: i would defer documenting this in docs.newrelic.com for now until we get feedback on our approach for otel + nr agent

**Review by bizob2828 — CHANGES_REQUESTED** [review]
overall looks great. called out a missing branch for testing

File: lib/config/index.js (hunk 2 above)
Comment (jordigh) [reply lib/config/index.js:1309]: Yeah, just like `NEW_RELIC_HOST`.

File: lib/config/default.js (hunk 3 above)
Comment (jordigh) [reply lib/config/default.js:61]: Do you mean I should delete this docstring?

**Review by bizob2828 — APPROVED** [review]

**Comment by bizob2828** [comment]
I cleared the proposed release notes. we can wordsmith this one later if at all

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**